From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
Subject: [PATCH 1/2] tc: add physics init/run/shutdown hooks into game loop

Adds optional TC physics calls in server init/frame/shutdown.

---
 src/game/g_main.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+ )

--- a/src/game/g_main.c
+++ b/src/game/g_main.c
@@
+#ifdef TC_PHYSICS
+extern void TC_Physics_Init(void);
+extern void TC_Physics_RunFrame(int msec);
+extern void TC_Physics_Shutdown(void);
+#endif
@@ void G_InitGame( int levelTime, int randomSeed, qboolean restart ) {
+#ifdef TC_PHYSICS
+    TC_Physics_Init();
+#endif
@@ void G_RunFrame( int levelTime ) {
+#ifdef TC_PHYSICS
+    TC_Physics_RunFrame( levelTime - level.previousTime );
+#endif
@@ void G_ShutdownGame( int restart ) {
+#ifdef TC_PHYSICS
+    TC_Physics_Shutdown();
+#endif
